src/core/global.c

```diff
@@ -1465,3 +1483,69 @@ void nn_global_rele_socket(struct nn_sock *sock)
     nn_sock_rele(sock);
     nn_glock_unlock();
 }
+
+int nn_setopt (int option, const void *optval, size_t optvallen)
+{
+    int idx;
+
+    switch (option) {
+    case NN_FORK_STRATEGY:
+        nn_assert(optvallen == sizeof (int));
+        idx = *(const int *) optval;
+        if (idx < 0 || idx >= NN_FORK_MAX_) {
+            errno = EINVAL;
+            break;
+        }
+#ifndef NN_HAVE_WINDOWS
+        nn_fork_strategy = &nn_fork_strategies[idx];
+#endif
+        return 0;
+    default:
+        errno = ENOPROTOOPT;
+        break;
+    }
```

Comment:
There really is no best strategy I'm afraid.

The current behaviour (aka `NN_FORK_NONE`) is fast and useful for children that call exec() after fork, while `NN_FORK_RESET` forces a reset of the library, can sometimes be fairly slow (and might even be slower if nanomsg implements multiple workers in the poller), and should really be a last-resort solution.

I was thinking of introducing a 3rd strategy after the initial work was done, that would give up on all pending I/O events and tasks in the child, effectively leaving the current opened sockets in a valid, non-working state. Admitedly, this could be more versatile than the two strategies introduced by this commit, but I wouldn't qualify it "better" in its own sense.